catalyst-gateway/bin/src/event_db/follower/mod.rs

```diff
@@ -0,0 +1,186 @@
+//! Follower Queries
+
+use cardano_chain_follower::Network;
+use handlebars::Handlebars;
+
+use crate::event_db::{Error, EventDB};
+
+/// Block time
+pub type DateTime = chrono::DateTime<chrono::offset::Utc>;
+/// Slot
+pub type SlotNumber = i64;
+/// Epoch
+pub type EpochNumber = i64;
+/// Block hash
+pub type BlockHash = String;
+/// Unique follower id
+pub type MachineId = String;
+
+/// `slot_no` column name
+const SLOT_NO_COLUMN: &str = "slot_no";
+/// `block_hash` column name
+const BLOCK_HASH_COLUMN: &str = "block_hash";
+/// `block_time` column name
+const BLOCK_TIME_COLUMN: &str = "block_time";
+/// `ended` column name
+const ENDED_COLUMN: &str = "ended";
+
+/// `insert_slot_index.sql`
+const INSERT_SLOT_INDEX_SQL: &str = include_str!("insert_slot_index.sql");
+/// `insert_update_state.sql`
+const INSERT_UPDATE_STATE_SQL: &str = include_str!("insert_update_state.sql");
+/// `select_update_state.sql`
+const SELECT_UPDATE_STATE_SQL: &str = include_str!("select_update_state.sql");
+/// `select_slot_info_by_datetime.sql.hbs`
+const SLOT_INFO_SQL_HBS: &str = include_str!("select_slot_info_by_datetime.sql.hbs");
+
+/// Query type
+pub(crate) enum SlotInfoQueryType {
+    /// Previous slot info query type
+    Previous,
+    /// Current slot info query type
+    Current,
+    /// Next slot info query type
+    Next,
+}
+
+/// Slot info query template fields
+#[derive(serde::Serialize)]
+struct SlotInfoQueryTmplFields {
+    /// `sign` field from the sql template
+    sign: &'static str,
+    /// `ordering` field from the sql template
+    ordering: Option<&'static str>,
+}
+
+impl SlotInfoQueryType {
+    /// Get SQL query
+    fn get_sql_query(&self) -> Result<String, Error> {
+        let tmpl_fields = match self {
+            SlotInfoQueryType::Previous => {
+                SlotInfoQueryTmplFields {
+                    sign: "<",
+                    ordering: Some("DESC"),
+                }
+            },
+            SlotInfoQueryType::Current => {
+                SlotInfoQueryTmplFields {
+                    sign: "=",
+                    ordering: None,
+                }
+            },
+            SlotInfoQueryType::Next => {
+                SlotInfoQueryTmplFields {
+                    sign: ">",
+                    ordering: None,
+                }
+            },
+        };
+
+        let mut reg = Handlebars::new();
+        // disable default `html_escape` function
+        // which transforms `<`, `>` symbols to `&lt`, `&gt`
+        reg.register_escape_fn(|s| s.into());
```

Comment:
This is fine, but its also able to be done with hndlebars itself.
If the template field is `{{{ something }}}` instead of just `{{ something }}` then handlebars will not do HTML escaping on that field.